lnst/Tests/Iperf.py

```diff
@@ -32,59 +32,47 @@ def run(self):
         server = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE,
                                   stderr=subprocess.PIPE, close_fds=True)
 
-        try:
-            stdout, stderr = server.communicate()
-            stdout = stdout.decode()
-            stderr = stderr.decode()
-        except KeyboardInterrupt:
-            pass
+        stdout, stderr = server.communicate()
+        stdout = stdout.decode().strip()
+        stderr = stderr.decode().strip()
+
+        if server.returncode > 0:
+            msg = f"iperf {self._role} returncode = {server.returncode}"
+            logging.error(msg)
+            logging.error(f"iperf stderr: {stderr}")
+            self._res_data["msg"] = msg
+            self._res_data["stderr"] = stderr
+            return False
 
         try:
             self._res_data["data"] = json.loads(stdout)
         except JSONDecodeError:
-            self._res_data["msg"] = "Error while parsing the iperf json output"
-            self._res_data["data"] = stdout
+            msg = "Error while parsing the iperf json output"
+            logging.error(msg)
+            logging.error(f"iperf stderr: {stderr}")
+            logging.debug(f"iperf stdout: {stdout}")
+            self._res_data["msg"] = msg
             self._res_data["stderr"] = stderr
-            logging.error(self._res_data["msg"])
             return False
 
-        try:
-            self._check_json_sanity()
-        except:
-            self._res_data["msg"] = "Iperf provided incomplete json data"
+        if not self._is_json_sane(self._res_data["data"]):
```

Comment:
Since we are renaming this - how about `_is_json_complete()`? The error message states "incomplete json data", so I think we could align the name with the message, wdyt?

Reply:
Also, the method takes `self._res_data["data"]` - why can't we change this call to:
```python
if not self._is_json_sane():
```
and make the method fetch it from `self._res_data["data"]`?

Otherwise it does not make sense to have this as a method since self is redundant.

Reply:
I don't think we should bind it to the internal implementation of the class. We're checking whether a json object returned by iperf is complete and that has nothing to do with the iperf arguments for example. I changed the method to be a `staticmethod` and changed the name to `_is_json_complete()`.

Reply:
I'm ok with staticmethod, resolving.